### PR TITLE
Add preliminary support for arm-windows and arm64-windows triplets

### PR DESCRIPTION
**Pull request by 4creators**
Visual Studio 15.4 shipped with new VC tools targeting arm and arm64
for desktop. This change allows for recognition and usage of new
triplets supporting arm and arm64 Windows desktop and server targets.

### CODE REVIEW

**Comment by 4creators** [comment]
This is related to  https://github.com/Microsoft/vcpkg/issues/2360

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -33,6 +35,8 @@ else()
         set(_VCPKG_TARGET_TRIPLET_ARCH x64)
     elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017 ARM")
         set(_VCPKG_TARGET_TRIPLET_ARCH arm)
+    elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017 ARM64")
```
Comment (4creators) [inline scripts/buildsystems/vcpkg.cmake]: Actually there is no CMake generator "Visual Studio 15 2017 ARM64", the generator  name is "Visual Studio 15 2017" but target architecture for generator is passed on CMake command line with `-A ARM64` switch. Do not know how to verify that "Visual Studio 15 2017" generator was accompanied with `-A ARM64` parameter to select `arm64` target.

File: ports/zlib/portfile.cmake [hunk 2]
```diff
@@ -13,6 +13,10 @@ vcpkg_apply_patches(
     ${CMAKE_CURRENT_LIST_DIR}/cmake_dont_build_more_than_needed.patch
 )
 
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" Or VCPKG_TARGET_ARCHITECTURE STREQAL "arm64")
+    if ()
+endif()
```
Comment (4creators) [inline ports/zlib/portfile.cmake]: WIP

File: scripts/buildsystems/vcpkg.cmake [hunk 3]
```diff
@@ -31,16 +33,20 @@ else()
         set(_VCPKG_TARGET_TRIPLET_ARCH x86)
     elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017 Win64$")
         set(_VCPKG_TARGET_TRIPLET_ARCH x64)
-    elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017 ARM")
+    elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017 ARM$")
         set(_VCPKG_TARGET_TRIPLET_ARCH arm)
-    elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017")
+    elseif(CMAKE_GENERATOR MATCHES "^Visual Studio 15 2017$" AND CMAKE_GENERATOR_PLATFORM MATCHES "^ARM64$")
```
Comment (4creators) [inline scripts/buildsystems/vcpkg.cmake]: CMake creates Visual Studio 2017 generator targeting Arm64 by using standard name without optional extension and passing -AARM64 command line switch to cmake.exe. This results in setting CMAKE_GENERATOR_PLATFORM to ARM64.

**Comment by 4creators** [comment]
@ras0219-msft

CMake missing feature allowing for Arm64 CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS support has been implemented and merged see: #2375 
arm-windows and arm64-windows targets should work for majority of ports

PTAL

File: scripts/cmake/vcpkg_configure_cmake.cmake [hunk 4]
```diff
@@ -59,7 +59,9 @@ function(vcpkg_configure_cmake)
 
     if(_csc_GENERATOR)
         set(GENERATOR ${_csc_GENERATOR})
-    elseif(_csc_PREFER_NINJA AND NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore" AND NOT _csc_HOST_ARCHITECTURE STREQUAL "x86")
+        # Ninja should not be used with Arm and Arm64 targets. Arm64 usage should be allowed
+        # once github issue #2375 is resolved 
+        elseif(_csc_PREFER_NINJA AND NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore" AND NOT _csc_HOST_ARCHITECTURE STREQUAL "x86" AND NOT VCPKG_TARGET_ARCHITECTURE MATCHES "^arm64$" AND NOT VCPKG_TARGET_ARCHITECTURE MATCHES "^arm$")
```
Comment (Mixaill) [inline scripts/cmake/vcpkg_configure_cmake.cmake]: indentation problem

**Comment by ras0219-msft** [comment]
Thanks for the PR! This is a great improvement and I'm really excited to see more ARM64 support for Windows :)

**Pull request merged**